Tests/MongoSwiftTests/MongoConnectionStringTests.swift

```diff
@@ -171,7 +171,8 @@ final class ConnectionStringTests: MongoSwiftTestCase {
         let encodedData = try JSONEncoder().encode(connStr)
         let decodedResult = try JSONDecoder().decode(MongoConnectionString.self, from: encodedData)
         expect(connStr.description).to(equal(decodedResult.description))
-        expect(connStr.description).to(equal("mongodb://localhost:27017/"))
+        // directConnection is always specified in URIs to override libmongoc behavior
+        expect(connStr.description).to(equal("mongodb://localhost:27017/?directconnection=false"))
```

Comment:
it's a little funny that it always shows up now when round tripping, though off the top of my head I can't think of any ill effects of that besides possibly a user finding it strange. 

what do you think about moving the defaulting logic into `withMongocURI`, and only setting it there for the purpose of libmongoc interop? then I think we wouldn't run into this issue, and it would be more obvious that we can delete the special logic once we stop interoperating with libmongoc

Reply:
good idea, updated.